src/mongoc/mongoc-topology-description.c

```diff
@@ -1370,6 +1370,46 @@ gSDAMTransitionTable[MONGOC_SERVER_DESCRIPTION_TYPES][MONGOC_TOPOLOGY_DESCRIPTIO
 /*
  *--------------------------------------------------------------------------
  *
+ * _mongoc_topology_description_type_string --
+ *
+ *      Get this topology's type, one of the types defined in the Server
+ *      Discovery And Monitoring Spec, as a string.
+ *
+ *      NOTE: this method should only be called while holding the mutex on
+ *      the owning topology object.
+ *
+ * Returns:
+ *       A string.
+ *
+ * Side effects:
+ *       None.
+ *
+ *--------------------------------------------------------------------------
+ */
+static const char *
+_mongoc_topology_description_type_string (mongoc_topology_description_t *topology)
```

Comment:
This is only for [CDRIVER-1198](https://jira.mongodb.org/browse/CDRIVER-1198). As I mentioned [below](https://github.com/mongodb/mongo-c-driver/pull/322/files#r59925833), the current trace lines require hoop-jumping to decipher, and a single log often has a bunch of those state changes. Using strings would be a huge convenience for debugging.